src/auth0-session/session/stateless-session.ts

```diff
@@ -0,0 +1,159 @@
+import * as jose from 'jose';
+import { CookieSerializeOptions, serialize } from 'cookie';
+import createDebug from '../utils/debug';
+import { Config } from '../config';
+import { Cookies } from '../utils/cookies';
+import { encryption } from '../utils/hkdf';
+import { AbstractSession, Header, SessionPayload } from './abstract-session';
+
+const debug = createDebug('stateless-session');
+
+const MAX_COOKIE_SIZE = 4096;
+const alg = 'dir';
+const enc = 'A256GCM';
+
+const notNull = <T>(value: T | null): value is T => value !== null;
+
+export class StatelessSession<
+  Req,
+  Res,
+  Session extends { [key: string]: any } = { [key: string]: any }
+> extends AbstractSession<Req, Res, Session> {
+  private keys?: Uint8Array[];
+
+  private chunkSize: number;
+
+  constructor(protected config: Config, protected Cookies: new () => Cookies) {
+    super(config, Cookies);
+    const {
+      cookie: { transient, ...cookieConfig },
+      name: sessionName
+    } = this.config.session;
+    const cookieOptions: CookieSerializeOptions = {
+      ...cookieConfig
+    };
+    if (!transient) {
+      cookieOptions.expires = new Date();
+    }
+
+    const emptyCookie = serialize(`${sessionName}.0`, '', cookieOptions);
+    this.chunkSize = MAX_COOKIE_SIZE - emptyCookie.length;
+  }
+
+  private async getKeys(): Promise<Uint8Array[]> {
```

Comment:
Should this method be in `AbstractSession`?

Reply:
The 2 session classes get different types of keys (signing for stateful and encrypting for stateless) so you can't share the code. And it's reasonable to expect that an implementation of abstract session might not need keys for either.